command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -56,29 +88,97 @@ func (c *cmd) Run(args []string) int {
 	}
 	defer f.Close()
 
-	meta, err := snapshot.Verify(f)
+	readFile, meta, err := snapshot.Read(nil, f)
 	if err != nil {
-		c.UI.Error(fmt.Sprintf("Error verifying snapshot: %s", err))
+		c.UI.Error(fmt.Sprintf("Error reading snapshot: %s", err))
+	}
+	stats, offset, err := enhance(readFile)
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error extracting snapshot data: %s", err))
 		return 1
 	}
 
-	var b bytes.Buffer
-	tw := tabwriter.NewWriter(&b, 0, 2, 6, ' ', 0)
-	fmt.Fprintf(tw, "ID\t%s\n", meta.ID)
-	fmt.Fprintf(tw, "Size\t%d\n", meta.Size)
-	fmt.Fprintf(tw, "Index\t%d\n", meta.Index)
-	fmt.Fprintf(tw, "Term\t%d\n", meta.Term)
-	fmt.Fprintf(tw, "Version\t%d\n", meta.Version)
-	if err = tw.Flush(); err != nil {
-		c.UI.Error(fmt.Sprintf("Error rendering snapshot info: %s", err))
+	//JSON and pretty print here
+	formatter, err := NewFormatter(c.format)
+	if err != nil {
+		c.UI.Error(err.Error())
 		return 1
 	}
+	metaformat := &MetadataInfo{
+		ID:      meta.ID,
+		Size:    meta.Size,
+		Index:   meta.Index,
+		Term:    meta.Term,
+		Version: meta.Version,
+	}
 
-	c.UI.Info(b.String())
-
+	in := &OutputFormat{
+		Meta:   metaformat,
+		Stats:  stats,
+		Offset: offset,
+	}
+	out, err := formatter.Format(in)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	c.UI.Output(out)
 	return 0
 }
 
+type typeStats struct {
+	Name       string
+	Sum, Count int
```

Comment:
Super nit picky but at least in the rest of Consul we don't seem to group struct fields like this. Function/Method returns and parameters is a whole other story though.

```suggestion
	Name  string
	Sum   int
	Count int
```